javaslang/src/main/java/javaslang/Lazy.java

```diff
@@ -135,14 +133,11 @@ private Lazy(Supplier<? extends T> supplier) {
      */
     @Override
     public T get() {
-        // using a local var speeds up the double-check idiom by 25%, see Effective Java, Item 71
-        Supplier<? extends T> tmp = supplier;
-        if (tmp != null) {
+        if ((value == null) && (supplier != null)) {
```

Comment:
Thank you for your expertise @shipilev!

As my comments also stated, I needed help in finding out why the obviously wrong code (i.e. no volatile) passed all the tests, before merging this PR (which wasn't obvious to me, I though the volatile read after the non-volatile one will synchronize both. My mistake.).

Will try to come up with a test that fails for the unsafe version you pointed out, so that I can better understand it - and we need to base our decisions on failing and passing tests anyway!

Thanks!